### ISSUE TITLE
Better error message for cf_cosine_plot when not enough neurons to cluster

### ISSUE THREAD

**Issue by mmc46** [natverse/coconatfly#20]
Co-clustering a type in 2 datasets, 2 neurons per dataset. One of neurons has hardly any connectivity, so using the default threshold of 5 generates an error that references the datasets. It would be helpful if the error message could make the issue clearer.

````
cf_cosine_plot(cf_ids(malecns = "type:DNge014",
                      flywire = "type:DNge014"),
               labRow = "{type}_{coconatfly::abbreviate_datasets(dataset)}{group}_{side}", 
               interactive = FALSE
, margins = c(5,5),
threshold = 5
)

Updating 6641 ids
Updating 5480 ids
Error in match.arg(ds, choices = cf_datasets("all"), several.ok = TRUE) : 
  'arg' must be of length >= 1
````

This works
```
cf_cosine_plot(cf_ids(malecns = "type:DNge014",
                      flywire = "type:DNge014"),
               labRow = "{type}_{coconatfly::abbreviate_datasets(dataset)}{group}_{side}", 
               interactive = FALSE
, margins = c(5,5),
threshold = 1
)
```

**Comment by jefferis** [comment]
Closed by #26

**Comment by jefferis** [comment]
see also #36 when there are no partners at all for either inputs/outputs
